src/main/java/org/jitsi_modified/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -248,73 +248,69 @@ else if (incomingBitrateInitialized)
-                    detector.estimator.getOffset(), tsDeltaMs,
-                    detector.estimator.getNumOfDeltas(), arrivalTimeMs);
-            }
+            detector.detector.detect(
+                detector.estimator.getOffset(), tsDeltaMs,
+                detector.estimator.getNumOfDeltas(), arrivalTimeMs);
+        }
+
+        // Check if it's time for a periodic update or if we
+        // should update because of an over-use.
+        if (lastUpdateMs == -1
+            || nowMs - lastUpdateMs > remoteRate.getFeedBackInterval())
+        {
+            updateEstimate = true;
+        }
+        else if (
+            detector.detector.getState() == BandwidthUsage.kBwOverusing)
+        {
+            long incomingRate_ = incomingBitrate.getRate(arrivalTimeMs);
 
-            // Check if it's time for a periodic update or if we
-            // should update because of an over-use.
-            if (lastUpdateMs == -1
-                || nowMs - lastUpdateMs > remoteRate.getFeedBackInterval())
+            if (incomingRate_ > 0 && remoteRate
+                .isTimeToReduceFurther(nowMs, incomingBitrate_))
             {
                 updateEstimate = true;
             }
-            else if (
-                detector.detector.getState() == BandwidthUsage.kBwOverusing)
-            {
-                long incomingRate_ = incomingBitrate.getRate(arrivalTimeMs);
-
-                if (incomingRate_ > 0 && remoteRate
-                    .isTimeToReduceFurther(nowMs, incomingBitrate_))
-                {
-                    updateEstimate = true;
-                }
-            }
+        }
 
-            if (updateEstimate)
-            {
-                // The first overuse should immediately trigger a new estimate.
-                // We also have to update the estimate immediately if we are
-                // overusing and the target bitrate is too high compared to
-                // what we are receiving.
-                input.bwState = detector.detector.getState();
-                input.incomingBitRate = incomingBitrate.getRate(arrivalTimeMs);
-                input.noiseVar = detector.estimator.getVarNoise();
-
-                remoteRate.update(input, nowMs);
-                targetBitrateBps = remoteRate.updateBandwidthEstimate(nowMs);
-                updateEstimate = remoteRate.isValidEstimate();
-            }
+        if (updateEstimate)
+        {
+            // The first overuse should immediately trigger a new estimate.
+            // We also have to update the estimate immediately if we are
+            // overusing and the target bitrate is too high compared to
+            // what we are receiving.
+            input.bwState = detector.detector.getState();
+            input.incomingBitRate = incomingBitrate.getRate(arrivalTimeMs);
+            input.noiseVar = detector.estimator.getVarNoise();
+
+            remoteRate.update(input, nowMs);
+            remoteRate.updateBandwidthEstimate(nowMs);
```

Comment:
Is it wrong we were never doing anything with the return value before?

Reply:
No, we used to fire an event, but we don't anymore:
https://github.com/jitsi/libjitsi/blob/master/src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java#L324